### PR TITLE
Minor tweaks to go CI

### PR DESCRIPTION
**Pull request by ghost**
## Scope

- Removing redundant `go vet` CI step
- Making sure go tooling is standardized on 1.17
- Moving to pre-built golangci-lint action

## Why?

- Less variability, faster builds, simpler CI.

## Todos

None.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -106,16 +106,6 @@ jobs:
         with:
           name: alicenet-${{ env.BRANCH_NAME }}-${{ env.OPERATING_SYSTEM }}
           path: ./madnet
-
```
Comment (vtleonardo) [inline .github/workflows/ci.yml:109]: Was go vet being executed somewhere else?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -125,11 +115,8 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - uses: ./.github/actions/alicenet-config
-      - name: install-golangci-lint
```
Comment (vtleonardo) [inline .github/workflows/ci.yml:128]: Maybe we should continue using the curl version here. We used the pre-compiled version of the golangci, some weeks ago. However, that was not displaying the errors correctly. Getting the version via curl solved some problems. The version with the problem:
https://github.com/MadBase/MadNet/runs/6392741284?check_suite_focus=true

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ghost) [reply .github/workflows/ci.yml:109]: golangci-lint includes `govet` as a default enabled linter.

Comment (vtleonardo) [reply .github/workflows/ci.yml:109]: Awesome!

Comment (vtleonardo) [reply .github/workflows/ci.yml:109]: Awesome!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (vtleonardo) [reply .github/workflows/ci.yml:128]: After discussion, we saw that this error may have been fixed on the latest version of the golinter Github's action.

**Review by vtleonardo — APPROVED** [review]

**Comment by ghost** [comment]
Thanks @vtleonardo!

**Pull request merged**